packages/api/endpoints/rules.js

```diff
@@ -121,9 +130,17 @@ async function put({ params: { name }, body }, res) {
     // since body is expected to be a replacement rule, not a partial rule
     const fieldsToDelete = Object.keys(oldRule).filter((key) => !(key in body));
     const newRule = await model.update(oldRule, body, fieldsToDelete);
+    const newPostgresRecord = await translateApiRuleToPostgresRule(newRule, dbClient);
 
     if (inTestMode()) await addToLocalES(newRule, indexRule);
 
+    await dbClient.transaction(async (trx) => {
```

Comment:
Using the transaction is only really useful if we're going to do multiple updates and we want to ensure that the Postgres update only happens if everything inside the transaction callback is successful. Thus, you need to move the `model.update` in this endpoint handler inside this transaction after the Postgres write. Then, if anything fails in the transaction, none of the writes happen so we have a guarantee that the update either happens in both Dynamo and Postgres or neither.

Reply:
[`ea9c778` (#2012)](https://github.com/nasa/cumulus/pull/2012/commits/ea9c7784b4d0e9841c8ebee053f05bbdcce89ce7)